lib/valueflow.cpp

```diff
@@ -3749,12 +3749,16 @@ struct ValueFlowConditionHandler {
 
                     // if astParent is "!" we need to invert codeblock
                     {
-                        const Token *parent = tok->astParent();
-                        while (parent && parent->str() == "&&")
-                            parent = parent->astParent();
-                        if (parent && (parent->str() == "!" || Token::simpleMatch(parent, "== false"))) {
-                            check_if = !check_if;
-                            check_else = !check_else;
+                        const Token *t = tok;
```

Comment:
I don't like the name `t`. It would be better with `tok2`. Or if you can think of something more descriptive feel free to change it. we do not limit number of characters in variable names.

Reply:
Done. Thanks for the review !